docs/rfcs/8-add-tcp-pair.md

```diff
@@ -0,0 +1,90 @@
+- Author: bokket  [bokkett@gmail.com](mailto:bokkett@gmail.com)
+- Start Date: 2021-07-17
+- RFC PR: [beyondstorage/go-endpoint#8](https://github.com/beyondstorage/go-endpoint/pull/8)
+- Tracking Issue: [beyondstorage/go-endpoint/issues/7](https://github.com/beyondstorage/go-endpoint/issues/7)
+
+# RFC-8: Add Tcp pair
+
+Releated issue: [beyondstorage/go-endpoint/issues/7](https://github.com/beyondstorage/go-endpoint/issues/7)
+
+## Background
+
+hdfs usually use the `New(address string)` method to access a namenode node, the user will be the user running the code. If the address is an empty string, it will try to get the NameNode address from the Hadoop configuration file.
+
+## Proposal
+
+I suggest adding a pair to allow the user to specify the address.
+
+- The `type` of `tcp` should be `String` and is a `const`
+- The `format` of `ProtocolTcp` should follow  [go-endpoint](https://github.com/beyondstorage/go-endpoint/blob/master/README.md)
+- The `value` of `endpoint` should be parsed into `address include host and port`
+- Now we don't have a pair operation on the `hdfs address` or tcp-like operation
+
+## Rationale
+
+### Why not use hadoop configuration?
+
+We can specify the configuration of the path via `LoadHadoopConf`
+If not specified, the default path `${hadoop_home}/conf` will be used
+
+However, there is no guarantee that the path is wrong, and it is more common for users to use the namenode address directly.
+
+## Compatibility
+
+No compatibility issues at this time.
+
+## Implementation
+
+First add a  `String` type with the name `ProtocolTcp` in `endpoint`. 
```

Comment:
We don't need to add so detailed code in RFC, instead, we can describe them in clear sentences.

For example:

- Add protocol `tcp`
- Implement protocol tcp formatted (`func (p Endpoint) Tcp() (address string)`)
- Implement protocol tcp parser (`func Parse(cfg string)`)